### PR TITLE
Avro serde support for .net standard 2.0

### PR DESCRIPTION
**Pull request by mhowlett**
this is dependent on merging https://github.com/confluentinc/avro/pull/5 . opening now to get a public CI build so people eager for this functionality can start testing it.

### CODE REVIEW

File: src/Confluent.Kafka/Serialization/Extensions/Message.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public static class MessageExtensions
             }
             catch (Exception ex)
             {
-                throw new KafkaException(new Error(ErrorCode.Local_KeyDeserialization, ex.Message), ex);
+                throw new KafkaException(new Error(ErrorCode.Local_KeyDeserialization, ex.ToString()), ex);
```
Comment (mhowlett) [inline src/Confluent.Kafka/Serialization/Extensions/Message.cs]: this provides a full stack trace rather than meaningless overview.

**Comment by mhowlett** [comment]
Depends on https://github.com/confluentinc/avro/pull/5

File: examples/AvroGeneric/AvroGeneric.csproj [hunk 2]
```diff
@@ -1,71 +1,12 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (edenhill) [inline examples/AvroGeneric/AvroGeneric.csproj]: This is nice, but is it safe with older .NET frameworks?

Comment (mhowlett) [reply examples/AvroGeneric/AvroGeneric.csproj]: yes, though probably not old versions of visual studio. but same constraints as other projects in the solution.

note: I'll do general verification of cross framework compatibility again when we do an RC.

**Pull request merged**